include/util/io.hpp

```diff
@@ -66,17 +56,14 @@ bool serializeVector(std::ostream &stream, const std::vector<simple_type> &data)
 template <typename simple_type>
 bool deserializeVector(const std::string &filename, std::vector<simple_type> &data)
 {
-    std::ifstream stream(filename, std::ios::binary);
 
-    if (!readAndCheckFingerprint(stream))
-        return false;
+    storage::io::File file(filename, true);
 
-    std::uint64_t count = 0;
-    stream.read(reinterpret_cast<char *>(&count), sizeof(count));
+    const auto count = file.readElementCount64();
     data.resize(count);
     if (count)
-        stream.read(reinterpret_cast<char *>(&data[0]), sizeof(simple_type) * count);
-    return static_cast<bool>(stream);
+        file.readInto(data.data(), count);
+    return true;
```

Comment:
If the function now throws on error, it should return `void` and not `bool`.

`bool fn();` makes it seem like the call site has to check for the return value, when it really does not.
Instead exceptions will be thrown and _that_ has to be accounted for.